fluss-client/src/test/java/org/apache/fluss/client/table/FlussTypedClientITCase.java

```diff
@@ -0,0 +1,452 @@
+        Integer a = i;
+        Boolean bool1 = (i % 2) == 0;
+        Byte tiny = (byte) (i - 5);
+        Short small = (short) (100 + i);
+        Integer intv = 1000 + i;
+        Long big = 100000L + i;
+        Float flt = 1.5f + i;
+        Double dbl = 2.5 + i;
+        Character ch = (char) ('a' + (i % 26));
+        String str = "s" + i;
+        byte[] bin = new byte[] {(byte) i, (byte) (i + 1), (byte) (i + 2)};
+        byte[] bytes = new byte[] {(byte) (10 + i), (byte) (20 + i)};
+        BigDecimal dec = new BigDecimal("12345." + (10 + i)).setScale(2, RoundingMode.HALF_UP);
+        LocalDate dt = LocalDate.of(2024, 1, 1).plusDays(i);
+        LocalTime tm = LocalTime.of(12, (i * 7) % 60, (i * 11) % 60);
+        LocalDateTime tsNtz = LocalDateTime.of(2024, 1, 1, 0, 0).plusSeconds(i).withNano(0);
+        Instant tsLtz = Instant.ofEpochMilli(1700000000000L + (i * 1000L));
+        return new AllTypesPojo(
+                a, bool1, tiny, small, intv, big, flt, dbl, ch, str, bin, bytes, dec, dt, tm, tsNtz,
+                tsLtz);
+    }
+
+    @Test
+    void testTypedAppendWriteAndScan() throws Exception {
+        // Build all-types log table schema
+        Schema schema =
+                Schema.newBuilder()
+                        .column("a", DataTypes.INT())
+                        .column("bool1", DataTypes.BOOLEAN())
+                        .column("tiny", DataTypes.TINYINT())
+                        .column("small", DataTypes.SMALLINT())
+                        .column("intv", DataTypes.INT())
+                        .column("big", DataTypes.BIGINT())
+                        .column("flt", DataTypes.FLOAT())
+                        .column("dbl", DataTypes.DOUBLE())
+                        .column("ch", DataTypes.CHAR(1))
+                        .column("str", DataTypes.STRING())
+                        .column("bin", DataTypes.BINARY(3))
+                        .column("bytes", DataTypes.BYTES())
+                        .column("dec", DataTypes.DECIMAL(10, 2))
+                        .column("dt", DataTypes.DATE())
+                        .column("tm", DataTypes.TIME())
+                        .column("tsNtz", DataTypes.TIMESTAMP(3))
+                        .column("tsLtz", DataTypes.TIMESTAMP_LTZ(3))
+                        .build();
+        TablePath path = TablePath.of("pojo_db", "all_types_log");
+        TableDescriptor td = TableDescriptor.builder().schema(schema).distributedBy(2).build();
+        createTable(path, td, true);
+
+        try (Table table = conn.getTable(path)) {
+            // write
+            TypedAppendWriter<AllTypesPojo> writer =
+                    table.newAppend().createTypedWriter(AllTypesPojo.class);
+            List<AllTypesPojo> expected = new ArrayList<>();
+            for (int i = 0; i < 5; i++) {
+                AllTypesPojo u = newAllTypesPojo(i);
+                expected.add(u);
+                writer.append(u).get();
```

Comment:
```suggestion
                writer.append(u);
```

use async flush to reduce test time